### ISSUE TITLE
v0.30: reimplement ip, ipv6 and bitrate (for wired connections) placeholders for net block

### ISSUE THREAD

**Issue by ammgws** [greshake/i3status-rust#1515]
I think ip/ipv6 can be done via netlink, not sure about bitrate though.

**Comment by ammgws** [comment]
This is probably the most important issue for releasing v0.3.

Bitrate could possibly be deprecated if not possible via netlink

**Comment by jcgruenhage** [comment]
Why would you go from 0.22.0 to 0.30.0?

**Comment by ammgws** [comment]
If you mean why the version was bumped to 0.30, it's because the core was completely rewritten and there are a lot of breaking changes.

**Comment by ammgws** [comment]
Yes if we were strictly following semver there would be no promise of stability but we've had an implied stability in that in general we have always tried not to break configs with releases. So when we had to implement major breaking changes bumping straight to the next minor version didn't seem like too weird.

**Comment by jcgruenhage** [comment]
Yes and no. The next minor version after 0.22.0 is 0.23.0. Considering that there have been breaking changes in both 0.22 and even 0.20.2, I would definitely not agree that there is any stability guarantee, neither implied nor perceived. Breaking changes in a patch version is a big no-no. If you want to indicate that releases are stable and want to make a bigger jump, why not just go to 1.0.0, with a few release candidates before to make sure it's going to go well? I'd say it would be fine honestly, and actually doing semantic versioning instead of indicating a bigger change by jumping to the next multiple of ten for the minor version would definitely increase the perceived stability of this project.

**Comment by MaxVerevkin** [comment]
The IP can be done via netlink, here is an example: https://github.com/jbaublitz/neli/blob/main/examples/route-list.rs

I'm not sure about IPv6 as the example seems to have some code to handle it, but it doesn't show it for me. We can look into `ip`'s source code to find how it finds it. Similarly, we can just look into `iw` to find how to deal with bitrate.

**Comment by AkechiShiro** [comment]
I've tried the 0.30 and right now, the brightness block when at the lowest percentage disappears from the bar (status command exited with an error code 101 I think then if you restart the i3status bar, you see the brightness block disappeared), you can't tweak it back unless you run a command in a terminal or using media Fn keys (if you binded them)

**Comment by MaxVerevkin** [comment]
@AkechiShiro please open a new issue

**Comment by MaxVerevkin** [comment]
> Why would you go from 0.22.0 to 0.30.0?

Apparently it is not uncommon for projects to skip a few versions to demonstrate fundamental changes. For example, https://github.com/Smithay/client-toolkit jumps from 0.16 to 0.30.

**Comment by MaxVerevkin** [comment]
I just realized that bitrate is wireless-specific and it is already available via `neli-wifi` (https://docs.rs/neli-wifi/latest/neli_wifi/struct.Station.html#structfield.tx_bitrate)

**Comment by MaxVerevkin** [comment]
By the way, why does the `net` block have only tx bitrate? What about rx bitrate? Dose it make sense to add another placeholder?

**Comment by ammgws** [comment]
I think because it was first added for wired connections. Since wifi has both, why not?

**Comment by MaxVerevkin** [comment]
Uhhh. Wired connection speed is available via `ethtool` protocol, which is supported by [`little-dude/netlink`](https://github.com/little-dude/netlink) but not `neli` while wireless bitrate is available via `nl80211` which is supported by `neli-wifi`, but not `little-dude/netlink` :sob:

**Comment by ammgws** [comment]
I think if anything supporting wifi makes sense since that's the one that varies

**Comment by MaxVerevkin** [comment]
Yeah, I guess so... On the side note, I like little-dude's netlink library API more, so I might end up rewriting neli-wifi.

**Comment by AkechiShiro** [comment]
The net block completely broke for me recently, was the latest version tested by anyone ?

The block fails to render, and it says `placeholder with name "frequency" not found` same with `ssid`.

My format for the `net` block is : 
```
format = "$ssid($frequency) $speed_down $speed_up" 
```

**Comment by MaxVerevkin** [comment]
> The block fails to render, and it says placeholder with name "frequency" not found same with ssid.

A few days ago SSID broke for me and I don't know why yet. Do run the latest commit? It _might_ at least fix frequency (it works for me).

**Comment by MaxVerevkin** [comment]
Downgrading to linux 5.18 fixes it. Interesting.

**Comment by AkechiShiro** [comment]
Is this a Rust bug ?

**Comment by MaxVerevkin** [comment]
> Is this a Rust bug ?

No. Either `linux 5.19` has a bug which breaks userspace (but this is not the case because everything else except `i3status-rs` rust works) or there is a bug in `neli-wifi` (most likely).

**Comment by MaxVerevkin** [comment]
Looks like it _is_ a linux bug. `iw dev` shows the SSID on `linux 5.18` but does not on `linux 5.19`.

**Comment by AkechiShiro** [comment]
Is this bug reported upstream or not ? Could anyone link it here ?

**Comment by ammgws** [comment]
> Uhhh. Wired connection speed is available via `ethtool` protocol, which is supported by [`little-dude/netlink`](https://github.com/little-dude/netlink) but not `neli` while wireless bitrate is available via `nl80211` which is supported by `neli-wifi`, but not `little-dude/netlink` sob

Hope for the future here? https://github.com/little-dude/netlink/pull/282

(although maybe not until https://github.com/little-dude/netlink/issues/257 is resolved)
